ebp/comcastebp.go

```diff
@@ -65,64 +66,62 @@ func (ebp *comcastEbp) SetDiscontinuityFlag(value bool) {
 }
 
 // readComcastEbp will parse raw bytes without the tag into a Comcast EBP.
-func readComcastEbp(data io.Reader) (ebp *comcastEbp, err error) {
+func readComcastEbp(data []byte) (ebp *comcastEbp, err error) {
 	ebp = &comcastEbp{
 		baseEbp: baseEbp{DataFieldTag: ComcastEbpTag},
 	}
 
-	if err = binary.Read(data, ebpEncoding, &ebp.DataFieldLength); err != nil {
-		return nil, err
+	// We read 2 bytes not based on flags
+	if len(data) < 2 {
+		return nil, gots.ErrNoPayload
 	}
 
-	remaining := ebp.DataFieldLength
+	index := uint8(0)
 
-	if remaining == 0 {
-		return ebp, nil
-	}
+	ebp.DataFieldTag = data[index]
+	index += uint8(1)
 
-	if err = binary.Read(data, ebpEncoding, &ebp.DataFlags); err != nil {
-		return nil, err
-	}
+	ebp.DataFieldLength = data[index]
+	index += uint8(1)
 
-	remaining -= uint8(1)
+	// Check if the data is as advertised
+	if ebp.DataFieldLength > 0 {
+		if len(data) >= 3 {
```

Comment:
Section 5.2 of the Comcast spec says we only need one byte (all flags) for a pedantically correct minimal EBP.

Granted, such an EBP would be useless because it would have all flags off and no time information.

Can probably get away w/o fixing this since this code is probably used rarely if at all.